js/rpg_core/Bitmap.js

```diff
@@ -41,14 +41,76 @@ function Bitmap() {
  *
  */
 
-Bitmap.prototype.initialize = function(width, height) {
-    this._canvas = document.createElement('canvas');
-    this._context = this._canvas.getContext('2d');
-    this._canvas.width = Math.max(width || 0, 1);
-    this._canvas.height = Math.max(height || 0, 1);
-    this._baseTexture = new PIXI.BaseTexture(this._canvas);
-    this._baseTexture.mipmap = false;
-    this._baseTexture.scaleMode = PIXI.SCALE_MODES.NEAREST;
+
+Bitmap.prototype._createCanvas = function(width, height){
+    this.__canvas = document.createElement('canvas');
+    this.__context = this.__canvas.getContext('2d');
+
+    this.__canvas.width = Math.max(width || 0, 1);
+    this.__canvas.height = Math.max(height || 0, 1);
+
+    if(this._image){
+        var w = Math.max(this._image.width || 0, 1);
+        var h = Math.max(this._image.height || 0, 1);
+        this.__canvas.width = w;
+        this.__canvas.height = h;
+
+        this._createBaseTexture(this._canvas);
+        this._baseTexture.width = w;
+        this._baseTexture.height = h;
+        this.smooth = this._smooth;
+
+        this.__context.drawImage(this._image, 0, 0);
+    }
+
+    this._setDirty();
+};
+
+Bitmap.prototype._createBaseTexture = function(source){
+    this.__baseTexture = new PIXI.BaseTexture(source);
+    this.__baseTexture.mipmap = false;
+    this.__baseTexture.scaleMode = PIXI.SCALE_MODES.NEAREST;
+};
+
+//
+//We don't want to waste memory, so creating canvas is deferred.
+//
+Object.defineProperties(Bitmap.prototype, {
+    _canvas: {
+        set: function(){
+            throw new Error('cannot set');
```

Comment:
If you wanna throw error when these properties are assigned, simply remove setter! (Also in 90, 100)
We should unify how to define properties with original corescript.